picard/script/__init__.py

```diff
@@ -101,3 +114,236 @@ def enabled_tagger_scripts_texts():
+        Properties in the JSON string that are not found in the script object are ignored.
+
+        Args:
+            json_string (str): JSON string containing the property settings.
+        """
+        settings = json.loads(json_string)
+        self._update_from_dict(settings)
+
+
+class FileNamingScript(PicardScript):
+    """Picard file naming script class
+    """
+    TYPE = PicardScriptType.FILENAMING
+    JSON_OUTPUT = ['title', 'script', 'author', 'description', 'license', 'version', 'last_updated']
+
+    def __init__(self, script='', title='', id=None, readonly=False, deletable=True, author='', description='', license='', version='', last_updated=None):
+        """Creates a Picard file naming script object.
+
+        Args:
+            script (str): Text of the script.
+            title (str): Title of the script.
+            id (str): ID code for the script. Defaults to a system generated uuid.
+            readonly (bool): Identifies if the script is readonly. Defaults to False.
+            deletable (bool): Identifies if the script can be deleted from the selection list. Defaults to True.
+            author (str): The author of the script. Defaults to ''.
+            description (str): A description of the script, typically including type of output and any required plugins or settings. Defaults to ''.
+            license (str): The license under which the script is being distributed. Defaults to ''.
+            version (str): Identifies the version of the script. Defaults to ''.
+            last_updated (str): The UTC date and time when the script was last updated. Defaults to current date/time.
+        """
+        super().__init__(script=script, title=title, id=id, last_updated=last_updated)
+        self.readonly = readonly    # for presets
+        self.deletable = deletable  # Allow removal from list of scripts
+        self.author = author
+        self.description = description
+        self.license = license
+        self.version = version
+
+    def copy(self):
+        new_object = super().copy()
+        new_object.readonly = False
+        new_object.deletable = True
+        return new_object
+
+
+def get_file_naming_script_presets():
+    """Generator of preset example file naming script objects.
+
+    Yields:
+        FileNamingScript: the next example FileNamingScript object
+    """
+    AUTHOR = "MusicBrainz Picard Development Team"
+    DESCRIPTION = _("This preset example file naming script does not require any special settings, tagging scripts or plugins.")
+    LICENSE = "GNU Public License version 2"
+
+    yield FileNamingScript(
+        title=_("Preset 1: Default file naming script"),
+        script=DEFAULT_FILE_NAMING_FORMAT,
+        readonly=True,
+        deletable=False,
+        author=AUTHOR,
+        description=DESCRIPTION,
+        version="1.0",
+        license=LICENSE,
+        last_updated="2019-08-05",
```

Comment:
Just a minor thing, but for consistency I would stick with the format defined at https://github.com/metabrainz/picard/pull/1810/files#diff-22922bde1619b8ae742614141b5fd647e7c9f973fee086f776e8d6c74212e37bR184

Reply:
I agree, it looks better being consistent.  Is the format okay?  I was wondering if it would make more sense showing local time (with the timezone).  Also not sure if we actually need the seconds precision.

Reply:
Dunno, I'm a bit bothered by this field, because it's complicated to maintain, perhaps it should be just handled automatically by Picard, if it has the specified format, and left untouched if not.
Frankly dunno at this point.
Seconds precision seems too much, but on the other side, that's pretty common datetime format.

Reply:
Actually, that's how I had it originally (readonly and automatically updated by Picard).  I can change it back to that easily enough, and drop the seconds resolution at the same time (although that came in handy during testing :wink: ).